cpp/src/cylon/io/arrow_io.cpp

```diff
@@ -35,25 +33,24 @@ arrow::Result<std::shared_ptr<arrow::Table>> read_csv(std::shared_ptr<cylon::Cyl
                                                       cylon::io::config::CSVReadOptions options) {
   arrow::Status st;
   auto *pool = cylon::ToArrowPool(ctx);
-  arrow::Result<std::shared_ptr<arrow::io::MemoryMappedFile>> mmap_result =
-      arrow::io::MemoryMappedFile::Open(path, arrow::io::FileMode::READ);
-  if (!mmap_result.status().ok()) {
+  const auto &mmap_result = arrow::io::MemoryMappedFile::Open(path, arrow::io::FileMode::READ);
+  if (!mmap_result.ok()) {
     return mmap_result.status();
   }
 
-  auto read_options = dynamic_cast<arrow::csv::ReadOptions *>(
+  auto *read_options = dynamic_cast<arrow::csv::ReadOptions *>(
```

Comment:
Only this one needs to be converted to a pointer?

Reply:
nice catch! that was a mistake. :D